packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -365,13 +375,17 @@ class AppBar extends StatelessWidget {
       );
     }
 
-    appBar = new Material(
-      color: backgroundColor ?? themeData.primaryColor,
-      elevation: elevation,
-      child: appBar
+    return new Hero(
+      tag: heroTag ?? _kDefaultHeroTag,
+      child: new Material(
+        color: backgroundColor ?? themeData.primaryColor,
+        elevation: elevation,
+        child: new Align(
+          alignment: FractionalOffset.topCenter,
+          child: appBar
```

Comment:
The appBar may contain elements that depend on InheritedWidgets; that may fail it's a hero: https://github.com/flutter/flutter/issues/213 (that's an old one)